src/test/scala/akka/stream/contrib/AccumulateWhileUnchangedSpec.scala

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.contrib
+
+import akka.stream.scaladsl.{ Keep, Source }
+import akka.stream.testkit.scaladsl.{ TestSink, TestSource }
+
+import scala.collection.immutable
+
+class AccumulateWhileUnchangedSpecAutoFusingOn extends {
+  val autoFusing = true
+} with AccumulateWhileUnchangedSpec
+
+class AccumulateWhileUnchangedSpecAutoFusingOff extends {
+  val autoFusing = false
+} with AccumulateWhileUnchangedSpec
+
+trait AccumulateWhileUnchangedSpec extends BaseStreamSpec {
+
+  "AccumulateWhileUnchanged" should {
+
+    "emit accumulated elements when the given property changes" in {
+      val (_, sink) = Source(SampleElements.All)
+        .via(AccumulateWhileUnchanged(_.value))
+        .toMat(TestSink.probe)(Keep.both)
```

Comment:
You can use `Keep.right` and store as `val sink = ...` instead of `val (_, sink) = ...`